lib/devise/models/trackable.rb

```diff
@@ -34,6 +34,13 @@ def update_tracked_fields!(request)
         update_tracked_fields(request)
         save
       end
+
+      protected
+
+        def get_ip(request)
```

Comment:
Thanks for the pull request @mckramer!

I have some minor comments, they're more about code style:

- Leave the method definition at the same indentation level as the `protected` keyword;
- Change the method name to `extract_ip_from`

Reply:
Thanks for the feedback.  I made the changes.

Reply:
> Thanks for the feedback. I made the changes.

Why did you choose to use 'protected' method instead of 'private' ?

Reply:
@kuzminvn2010 The idea was to able to override this method.

Reply:
Hi @tegon
But you can also override the private method.  No worse than protected.

```
module A
  def public_a_in_a
    a
  end

  private

  def a
    1
  end
end

class B
  include A
  
  def public_a_in_b
    a
  end

  private

  def a
    super + 1
  end
end

B.new.public_a_in_a #=> 2
B.new.public_a_in_b #=> 2
```

Reply:
@tegon ping

Reply:
Even though you're able to, it shows our intention. If we had chosen to make it `private`, that would mean that is a method people should be careful when using it, because it can change in patch and minor versions.